src/__tests__/search-and-replace-test.jsx

```diff
@@ -0,0 +1,423 @@
+var assert = require("assert");
+var StatefulEditorPage = require("../stateful-editor-page.jsx");
+var EditorPage = require("../editor-page.jsx");
+var ArticleEditor = require("../article-editor.jsx");
+var TestUtils = React.addons.TestUtils;
+
+describe("Search and Replace", function() {
+
+    describe("EditorPage", function() {
+        var editor;
+        var searchInput, replaceInput;
+        var previousButton, nextButton, replaceButton, replaceAllButton;
+
+        // Note: the example question initially was using expression widgets but
+        // this caused lots of issues because this widget type uses MathJax which
+        // isn't loaded in the testing environment.
```

Comment:
This took a while to figure out.  It worked in isolation for some reason but when running the tests together it failed.